pkg/storage/storage.go

```diff
@@ -1084,6 +1091,13 @@ func (is *ImageStore) checkCacheBlob(digest string) (string, error) {
 		return "", err
 	}
 
+	// there is an entry of digest in cache with empty value.
+	if dstRecord == "" {
```

Comment:
empty record means "no entry found", would enforce that check in cache.PutBlob() that we don't ever write an empty record

Reply:
Yes we can add additional check there also